solutions/src/main/java/ru/sberbank/school/task10/ScalableThreadPool.java

```diff
@@ -0,0 +1,105 @@
+package ru.sberbank.school.task10;
+
+import lombok.NonNull;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+public class ScalableThreadPool implements ThreadPool {
+    private final Queue<Runnable> taskQueue;
+
+    private final int min;
+    private final int max;
+    private volatile int workingThreads = 0;
+    private final List<ThreadWorker> threads;
+    private final Queue<ThreadWorker> freeThreads;
+
+    public ScalableThreadPool(int min, int max) {
+        if (min <= 0 || max <= 0 || min > max) {
+            throw new IllegalArgumentException();
+        }
+        this.taskQueue = new LinkedList<>();
+        this.min = min;
+        this.max = max;
+        this.threads = new ArrayList<>(max);
+        freeThreads = new LinkedList<>();
+        for (int i = 0; i < max; i++) {
+            ThreadWorker threadWorker = new ThreadWorker(i);
+            threads.add(threadWorker);
+            freeThreads.add(threadWorker);
+        }
+    }
+
+    @Override
+    public void start() {
+        synchronized (freeThreads) {
+            for (int i = 0; i < min; i++) {
+                    ThreadWorker threadWorker = freeThreads.poll();
```

Comment:
&#39;for&#39; child has incorrect indentation level 20, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1528448322>*